### PR TITLE
Add gcp secrets manager support

### PR DESCRIPTION
**Pull request by kimond**
Before opening a PR, make sure that, if relevant:
- [x] Changes are covered by automated tests.
- [ ] New / updated functions / methods have typehints.
- [ ] READMEs have been updated.
- [ ] The affected parties are / will be notified of incoming breaking changes.

### CODE REVIEW

File: opset/gcp_secret_handler.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+
+OPSET_GCP_PREFIX = "opset+gcp://"
+
+
+def is_gcp_available() -> bool:
+    return _has_secretmanager
+
+
+def retrieve_gcp_secret_value(secret_string: str) -> Any:
```
Comment (jvlar) [inline opset/gcp_secret_handler.py]: str

File: opset/gcp_secret_handler.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        tokens = path.split("/")
+
+        if f"{prefix}//" != OPSET_GCP_PREFIX or tokens[0] != "projects" or tokens[2] != "secrets":
+            raise InvalidGcpSecretStringException(secret_string)
+    except IndexError:
+        raise InvalidGcpSecretStringException(secret_string)
+
+
+def _add_version_if_needed(secret_name: str) -> str:
+    if "versions" not in secret_name:
```
Comment (jvlar) [inline opset/gcp_secret_handler.py]: will fail if versions is in the project or secret name

File: opset/configurator.py [hunk 3]
```diff
@@ -559,3 +618,26 @@ def load_logging_config(
...
+
+    :return: Opset Config or None
+    """
+    current_dir = os.path.dirname(__file__)
+    config_path = _search_for_config_file(current_dir)
+
+    if config_path:
+        with open(config_path, "r") as config_file:
+            return yaml.load(config_file, Loader=yaml.FullLoader) or {}
+
```
Comment (MarcDufresne) [inline opset/configurator.py:635]: I'd add a `return None` here just to be explicit

File: opset/configurator.py [hunk 4]
```diff
@@ -559,3 +618,26 @@ def load_logging_config(
             logging.getLogger(logger_name).setLevel(min_level)
 
     return root_logger
+
+
+def get_opset_config() -> dict[str, Any] | None:
+    """Search for opset config and load it if it exists.
+
+    :return: Opset Config or None
```
Comment (MarcDufresne) [inline opset/configurator.py]: You have a mix of Sphinx and Google style docstrings, I don't remember which is used by Opset but it should probably the same everywhere

File: tests/configurator_test.py [hunk 5]
```diff
@@ -280,3 +284,45 @@ def test_json_format() -> None:
...
+        assert config.secret == json_loaded_value
+        assert config.app.api_key == json_loaded_value
+
+
+def test_get_opset_config(mocker: MockerFixture) -> None:
+    mocker.patch(f"{TESTING_MODULE}.os.path.exists", return_true=True)
+    mocker.patch("builtins.open")
+    mock_yaml = mocker.patch(f"{TESTING_MODULE}.yaml")
+    fake_config = mocker.MagicMock()
+    fake_config.configure_mock(gcp_project_mapping={"test-1991": "test"})
```
Comment (MarcDufresne) [inline tests/configurator_test.py]: Technically it works like this, but shouldn't it be the opposite to reflect a normal mapping? When you have a "clean" name (`test`), that maps to a unique ID in GCP (`test-1991`)

File: tests/gcp_secret_handler_test.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+@pytest.fixture()
+def mock_access_secret_version(mocker: MockerFixture):
+    mock_client = mocker.patch(f"{TESTING_MODULE}.secretmanager.SecretManagerServiceClient")
+    mock_access_secret_version = mocker.MagicMock()
+    mock_client.return_value.access_secret_version = mock_access_secret_version
+
+    return mock_access_secret_version
+
+
+def _mock_gcp_responGse() -> MagicMock:
```
Comment (MarcDufresne) [inline tests/gcp_secret_handler_test.py]:
```suggestion
def _mock_gcp_response() -> MagicMock:
```

File: opset/configurator.py (hunk 4 above)
Comment (kimond) [reply opset/configurator.py]: Intellij auto complete madness :upside_down_face:

**Review by MarcDufresne — APPROVED** [review]

**Pull request merged**